pippo-core/src/main/java/ro/fortsoft/pippo/core/Response.java

```diff
@@ -576,7 +576,7 @@ public void send(CharSequence content) {
     public void resource(InputStream input) {
         checkCommitted();
 
-        if (isHeaderEmpty(HttpConstants.Header.CONTENT_TYPE)) {
+        if (isHeaderEmpty(HttpConstants.Header.CONTENT_TYPE) && getHttpServletResponse().getContentType() == null) {
             header(HttpConstants.Header.CONTENT_TYPE, HttpConstants.ContentType.APPLICATION_OCTET_STREAM);
         }
 
```

Comment:
`Content-Length` is in a funny situation right now.  This is directly related to flushing in `commit()`.

`Content-Length` is an HTTP 1.0 header and it is based on sending the entire response in one shot.  When you flush the response in `commit()` that changes things and sets transfer encoding to *chunked* which is a streaming mode for HTTP 1.1.  This lets the container buffer the response and send pieces over the connection.  When the container is chunking, I believe it removes the `Content-Length` header from the response.  So at the moment I don't think there is any value to this method.